<launch>

    <arg name="physical_robot" default="true"/>
    <arg name="use_joy" default="true"/>


    <!-- Simulation Parameter -->

    <node pkg="geosacs" type="visualisation_rviz.py" name="visualisation_rviz" output="screen"/>
    <arg name="output" default="/ik_interface/joint_states_sim"/>

    <!-- Load URDF -->
    <param name="robot_description" textfile="$(find lio_description)/urdf/lio.urdf" />
    <!-- Launch RVIZ -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find geosacs)/rviz/geosacs.rviz" />
    <arg name="node_start_delay" default="1.0" />  
    <!-- Launch robot state publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="lio_state_publisher" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " output="screen">
        <remap from="joint_states" to="$(arg output)"/>
        <param name="publish_frequency" type="double" value="1000.0" />
    </node>

  <group if="$(arg physical_robot)">
    <node pkg="myp_app_manager" type="myp_manager.py" name="myp_manager" output="screen"/>
    <node pkg="myp_app_manager" type="myp_application.py" name="myp_application" output="screen"/>
  </group> 

    <group if="$(arg use_joy)">
      <rosparam param="joy_node/dev">"/dev/input/js0"</rosparam>
      <include file="$(find teleop_twist_joy)/launch/teleop.launch">  
        <arg name="joy_dev" value="/dev/input/js0" />
      </include>
    </group>


</launch>